tools/azure-sdk-tools/devtools_testutils/resource_testcase.py

```diff
@@ -101,6 +101,9 @@ def create_resource(self, name, **kwargs):
 
     def remove_resource(self, name, **kwargs):
         if self.is_live and self._need_creation:
+            name = u"rgpy-" + name
```

Comment:
Oh this must have broken ever since you added the prefix. I was under an incorrect assumption when I responded to issue I thought they were hardcoding something in the test code.

Reply:
I don't think adding this hardcode tricky logic in both `create_resource` and `remove_resource` is a good way. Maybe you can consider adding prefix for the `name` before you pass `name` parameter into `create_resource` and `remove_resource`.

Reply:
Also the log shows the wrong name (the name without prefix), actually we create resource with prefix.
```Python
                logging.info(
                    "Attempting to create a Resource Group with name {} and parameters {}".format(name, parameters)
                )
                # Prefixing all RGs created here with 'rgpy-' for tracing purposes
                name = u"rgpy-" + name
                if len(name) > 90:
                    name = name[:90]
                self.resource = self.client.resource_groups.create_or_update(name, parameters)
```

Reply:
> I don't think adding this hardcode tricky logic in both `create_resource` and `remove_resource` is a good way. Maybe you can consider adding prefix for the name before you pass name parameter into `create_resource` and `remove_resource`.

@jsntcy I don't understand what you mean by this, can you elaborate or provide a code snippet?

Reply:
I'm not sure we would want to update all the callers of create/remove resources to have them pass in the name with the prefix but I do think it is a good idea to factor the name updates into a common function that we call at the beginning of both create and remove resources. 

Also from a quick look it seems as though there is already a name_prefix value in this class which serves a slight different but similar purpose. We might want to reconcile the concepts and perhaps have a rgname_prefix or something along those lines.

Reply:
Yes, agree with @weshaggard, since both `create_resource` and `remove_resource` are in class `ResourceGroupPreparer`, you can define a variable in class which can be shared among all functions in class `ResourceGroupPreparer`.

Not sure if we can reuse `self.name_prefix` which already exists in class `ResourceGroupPreparer`.
https://github.com/Azure/azure-sdk-for-python/blob/master/tools/azure-sdk-tools/devtools_testutils/resource_testcase.py#L62

Reply:
@jsntcy a few weeks ago our Subscription ID hit the maximum number of resource groups allowed, and we had no information on hundreds of RGs. Wes and I added the prefix so we could easily determine where at  least some of the generated RGs were coming from.